### PR TITLE
Avoid importing Qt when running "import mantid"

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Some python algorithms import the reducer framework when they load. The reducer framework currently has a `mantidqt` dependency and forces the Qt libraries to load then the framework `mantid` module is loaded. This work removes that dependency by delaying the import until it is needed.

It also forces the PyQt api to use version 2 for all testing so that it is compatible with `qtpy` or else you might see the error "PyQt4.QtCore and PyQt5.QtCore modules both wrap the QObject class". This seems to be particularly problematic on macOS.

**Report to:** [nobody]. 

**To test:**

Check an interface such as "DGS Reduction" with the CNCS data in the system test data directory. Simply enter some sample data and click run - it should reduce successfully. Check this in both mantidplot and workbench.

*There is no associated issue.*

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by samueljackson92** [comment]
I get the following message when I try to open CSNS data in DGS Reduction

```
objc[92642]: Class QCocoaMenuLoader is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaeb20) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba32c0). One of the two will be used. Which one is undefined.
objc[92642]: Class QNSApplication is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaea80) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba3310). One of the two will be used. Which one is undefined.
objc[92642]: Class QCocoaApplicationDelegate is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaea30) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba3360). One of the two will be used. Which one is undefined.
objc[92642]: Class QNSOpenSavePanelDelegate is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edae760) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba34a0). One of the two will be used. Which one is undefined.
objc[92642]: Class QCocoaPageLayoutDelegate is implemented in both /usr/local/opt/qt/lib/QtPrintSupport.framework/Versions/5/QtPrintSupport (0x108376ef0) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba35e0). One of the two will be used. Which one is undefined.
objc[92642]: Class QCocoaPrintPanelDelegate is implemented in both /usr/local/opt/qt/lib/QtPrintSupport.framework/Versions/5/QtPrintSupport (0x108376f40) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba3658). One of the two will be used. Which one is undefined.
objc[92642]: Class QNSImageView is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaec10) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba3680). One of the two will be used. Which one is undefined.
objc[92642]: Class QNSStatusItem is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaec60) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba36d0). One of the two will be used. Which one is undefined.
objc[92642]: Class QNSMenu is implemented in both /usr/local/Cellar/qt/5.9.2/plugins/platforms/libqcocoa.dylib (0x11edaecb0) and /usr/local/opt/qt@4/lib/QtGui.framework/Versions/4/QtGui (0x128ba36f8). One of the two will be used. Which one is undefined.
Traceback (most recent call last):
  File "/Users/samueljackson/git/mantid-fork-3/scripts/Interface/reduction_gui/widgets/inelastic/dgs_sample_setup.py", line 143, in _sample_browse
    fname = self.data_browse_dialog()
  File "/Users/samueljackson/git/mantid-fork-3/scripts/Interface/reduction_gui/widgets/base_widget.py", line 157, in data_browse_dialog
    flist = QFileInfo(flist).filePath()
TypeError: arguments did not match any overloaded call:
  QFileInfo(): too many arguments
  QFileInfo(str): argument 1 has unexpected type 'list'
  QFileInfo(QFile): argument 1 has unexpected type 'list'
  QFileInfo(QDir, str): argument 1 has unexpected type 'list'
  QFileInfo(QFileInfo): argument 1 has unexpected type 'list'
[1]    92642 abort      ./bin/workbench
```

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
See message above.

**Comment by peterfpeterson** [comment]
@samueljackson92's issue is unrelated, but needs a fix.

**Comment by martyngigg** [comment]
Working on it

**Comment by martyngigg** [comment]
The error should be fixed now. Once the mac packaging goes through we should probably put an abstraction in the way for the `QFileDialog` stuff so everyone doesn't have to cope with this.

File: MantidPlot/src/main.cpp [hunk 1]
```diff
@@ -154,6 +154,11 @@ the indentation depth for him/herself.
 */
 
 int main(int argc, char **argv) {
+  // Force the qtpy package to use the PyQt4 bindings. QtPy < v1.4 does not
+  // check whether a particular binding is imported and selects PyQt5 by default
+  // causing MantidPlot to segfault.
+  qputenv("QT_API", "pyqt");
```
Comment (peterfpeterson) [inline MantidPlot/src/main.cpp]: Please remove it from `buildconfig/CMake/Packaging/launch_mantidplot.sh.in` so it isn't set in two places

Comment (martyngigg) [reply MantidPlot/src/main.cpp]: :+1:

**Comment by samueljackson92** [comment]
In workbench the DGSRedution GUI now opens the file. Hitting reduce loads the data and kicks off the `DgsReduction` algorithm but then crashes with the following message:

```
...
DgsConvertToEnergyTransfer-[Notice] Starting DgsConvertToEnergyTransfer
DgsConvertToEnergyTransfer-[Notice] Processing for SNS
DgsConvertToEnergyTransfer-[Notice] Trying to determine file name
DgsConvertToEnergyTransfer-[Error] Error in execution of algorithm DgsConvertToEnergyTransfer:
DgsConvertToEnergyTransfer-[Error] Algorithm not registered (empty algorithm name)
DgsReduction-[Error] Error in execution of algorithm DgsReduction:
DgsReduction-[Error] Algorithm not registered (empty algorithm name)
scripter-[Error] Failed to execute script: RuntimeError on line 1102 of '/Users/samueljackson/builds/mantid-fork-3/release/bin/mantid/simpleapi.py': Algorithm not registered (empty algorithm name)
Traceback (most recent call last):
  File "/Users/samueljackson/git/mantid-fork-3/scripts/Interface/reduction_gui/reduction/scripter.py", line 666, in on_error
    raise RuntimeError(str(arg))
RuntimeError: RuntimeError on line 1102 of '/Users/samueljackson/builds/mantid-fork-3/release/bin/mantid/simpleapi.py': Algorithm not registered (empty algorithm name)
[1]    29013 abort      ./bin/workbench
```

**Comment by martyngigg** [comment]
@samueljackson92 I think this is actually an unrelated error on mac for DgsReduction. If the interface starts in both MantidPlot and Workbench then this issue is resolved.

You could also check the following:

**Comment by martyngigg** [comment]
Start a vanilla python instance
```
import mantid.simpleapi
import sys

print('PyQt4' in sys.modules) 
print('PyQt5' in sys.modules)
``
should print
```
False
False
```

I'll make a separate issue for the DgsReduction in workbench issue.

**Review by samueljackson92 — APPROVED** [review]
@martyngigg That works in vanilla python. I can see the other issue is unrelated, so I think this is good to go.

**Comment by martyngigg** [comment]
@mantidproject/gatekeepers Would we be able to get this merged? I'd like to check #23875 against it.

**Comment by martyngigg** [comment]
Thanks!

**Pull request merged**